File: Desafios Aleatorios/Python/exe02.py
SENHA_REGISTRADA = "senha"

def obter_senha() -> str:
    return input("Informe a senha: ")

def validar_senha(senha: str) -> bool:
    return senha == SENHA_REGISTRADA

def liberar_acesso(tentativas: int) -> bool:
    while tentativas > 0:
        print(f"Tentativas Restantes: {tentativas}")
        senha = obter_senha()
        if validar_senha(senha):
            print("Acesso Concedido")
            return True
        tentativas -= 1
    print("Acesso negado")
    return False

if __name__ == "__main__":
    acesso_liberado = liberar_acesso(tentativas=3)

File: Desafios Aleatorios/Java/exe02.java
import java.util.Scanner;

public class exe02 {
    public static void main(String[] args) {

        String SENHAREGISTRADA = "senha";
        String senhaUsuario; // obter a senha do usuario
        int tentativas = 3; // numero de tentativas

        // iniciar scanner
        Scanner scanner = new Scanner(System.in);

        // iniciar logica de verificação
        while (tentativas > 0) {
            System.out.println("Tentativas restantes: " + tentativas);
            System.out.print("Informe a senha: ");
            senhaUsuario = scanner.nextLine();
            if (!senhaUsuario.equals(SENHAREGISTRADA)){
                System.out.println("Senha Invalida tente novamente");
                tentativas--;
            } else {
                break;
            }
        }
        
        if (tentativas > 0){
            System.out.println("Acesso concedido");
        } else {
            System.out.println("Acesso bloqueado");
        }

        // fechar scanner
        scanner.close();
    }

}
